code/datums/gamemode/factions/bloodcult/bloodcult_mobs_and_constructs.dm

```diff
@@ -0,0 +1,439 @@
+			qdel(obstacle)
+			breakthrough = 1
+
+		else if(istype(obstacle, /turf/simulated/wall))
+			var/turf/simulated/wall/W = obstacle
+			if (W.hardness <= 60)
+				playsound(W, 'sound/weapons/heavysmash.ogg', 75, 1)
+				W.dismantle_wall(1)
+				breakthrough = 1
+			else
+				src.throwing = 0
+				src.crashing = null
+
+		else if(istype(obstacle, /obj/structure/reagent_dispensers/fueltank))
+			obstacle.ex_act(1)
+
+		else if(istype(obstacle, /mob/living))
+			var/mob/living/L = obstacle
+			if (L.flags & INVULNERABLE)
+				src.throwing = 0
+				src.crashing = null
+			else if (!(L.status_flags & CANKNOCKDOWN) || (M_HULK in L.mutations) || istype(L,/mob/living/silicon))
+				//can't be knocked down? you'll still take the damage.
+				src.throwing = 0
+				src.crashing = null
+				L.take_overall_damage(5,0)
+				if(L.locked_to)
+					L.locked_to.unlock_atom(L)
+			else
+				L.take_overall_damage(5,0)
+				if(L.locked_to)
+					L.locked_to.unlock_atom(L)
+				L.Stun(5)
+				L.Knockdown(5)
+				L.apply_effect(STUTTER, 5)
+				playsound(src, 'sound/weapons/heavysmash.ogg', 50, 0, 0)
+				breakthrough = 1
+		else
+			src.throwing = 0
+			src.crashing = null
+
+		if(breakthrough)
+			if(crashing && !istype(crashing,/turf/space))
+				spawn(1)
+					src.throw_at(crashing, 50, src.throw_speed)
+			else
+				spawn(1)
+					crashing = get_distant_turf(get_turf(src), dash_dir, 2)
+					src.throw_at(crashing, 50, src.throw_speed)
+
+	if(istype(obstacle, /obj))
+		var/obj/O = obstacle
+		if(!O.anchored)
+			step(obstacle,src.dir)
+		else
+			obstacle.Bumped(src)
+	else if(istype(obstacle, /mob))
+		step(obstacle,src.dir)
+	else
+		obstacle.Bumped(src)
+
+
+////////////////////Wraith/////////////////////////
+
+
+/mob/living/simple_animal/construct/wraith/perfect
+	icon_state = "wraith2"
+	icon_living = "wraith2"
+	icon_dead = "wraith2"
+	see_in_dark = 7
+	construct_spells = list(/spell/targeted/ethereal_jaunt/shift/alt)
+	var/ranged_cooldown = 0
+	var/ammo = 3
+	var/ammo_recharge = 0
+
+/mob/living/simple_animal/construct/wraith/perfect/New()
+	..()
+	setupfloat()
+
+/mob/living/simple_animal/construct/wraith/perfect/Life()
+	if(timestopped)
+		return 0
+	. = ..()
+	ranged_cooldown = max(0,ranged_cooldown-1)
+	if (ammo < 3)
+		ammo_recharge++
+		if (ammo_recharge >= 3)
+			ammo_recharge = 0
+			ammo++
+
+/mob/living/simple_animal/construct/wraith/perfect/RangedAttack(var/atom/A, var/params)
+	if(ranged_cooldown <= 0 && ammo)
+		ammo--
+		generic_projectile_fire(A, src, /obj/item/projectile/wraithnail, 'sound/weapons/hivehand.ogg')
+	return ..()
+
+/obj/item/projectile/wraithnail
+	icon = 'icons/obj/projectiles_experimental.dmi'
+	icon_state = "wraithnail"
+	damage = 5
+
+
+/obj/item/projectile/wraithnail/to_bump(var/atom/A)
+	if(bumped)
+		return 0
+	bumped = 1
+
+	if(A)
+		setDensity(FALSE)
+		invisibility = 101
+		kill_count = 0
+		var/obj/effect/overlay/wraithnail/nail = new (A.loc)
+		nail.transform = transform
+		if(isliving(A))
```

Comment:
This all feels like it'll get pretty bullshit pretty fast. Why is armor not taken into account at all when dealing with this nailgun?

Reply:
The damage is negligible, especially when it can only fire once every 5 or so seconds. The point is that you'll take low damage now matter how armored you are.

Reply:
It doesn't stun, as in the person can still use items, fire guns, speak, etc. The nail can be removed after 1 second by clicking on it. Or just dodged, since it moves as slowly as a bullet.